### PR TITLE
feat(ci): Integrate CodeQL, Dependabot, Scorecard, and Dependency Reviews

### PR DESCRIPTION
**Pull request by JuanAgudeloRSL**
Set up repository maintenance and security tools:
- CodeQL: For code analysis and vulnerability detection.
- Dependabot: Automated dependency updates.
- Scorecard: Repository security evaluation.
- Dependency Reviews: Analyzes dependency risks for security and licensing.

### CODE REVIEW

**Review by ezequiel-rodriguez — APPROVED** [review]
LGTM

File: .github/workflows/dependencies-review.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      pull-requests: write
+    steps:
+      - name: 'Checkout Repository'
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: 'Dependency Review'
+        uses: actions/dependency-review-action@3b139cfc5fae8b618d3eae3675e383bb1769c019 # v4.5.0
+        with:
+          fail-on-severity: high
+          comment-summary-in-pr: true
```
Comment (diego-jeronymo) [inline .github/workflows/dependencies-review.yml:20]: Please add `verbose: false`

Comment (ezequiel-rodriguez) [inline .github/workflows/dependencies-review.yml:20]:
```suggestion
          comment-summary-in-pr: true
          verbose: false
```

File: .github/workflows/dependencies-review.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+          verbose: false
```
Comment (diego-jeronymo) [inline .github/workflows/dependencies-review.yml:21]: please remove `verbose: false`

File: .github/workflows/dependencies-review.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      pull-requests: write
+    steps:
+      - name: 'Checkout Repository'
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: 'Dependency Review'
+        uses: actions/dependency-review-action@3b139cfc5fae8b618d3eae3675e383bb1769c019 # v4.5.0
+        with:
+          fail-on-severity: high
+          comment-summary-in-pr: true
```
Comment (diego-jeronymo) [inline .github/workflows/dependencies-review.yml:20]: please change this config `comment-summary-in-pr: on-failure`

Comment (ezequiel-rodriguez) [inline .github/workflows/dependencies-review.yml:20]:
```suggestion
          comment-summary-in-pr: on-failure
```

File: .github/workflows/dependencies-review.yml (hunk 2 above)
Comment (ezequiel-rodriguez) [inline .github/workflows/dependencies-review.yml:21]:
```suggestion
```

File: .github/workflows/dependencies-review.yml (hunk 1 above)
Comment (ezequiel-rodriguez) [inline .github/workflows/dependencies-review.yml:20]:
```suggestion
          comment-summary-in-pr: on-failure
```

File: .github/workflows/dependencies-review.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      pull-requests: write
+    steps:
+      - name: 'Checkout Repository'
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: 'Dependency Review'
+        uses: actions/dependency-review-action@3b139cfc5fae8b618d3eae3675e383bb1769c019 # v4.5.0
+        with:
+          fail-on-severity: high
+          comment-summary-in-pr: on-failure
```
Comment (ezequiel-rodriguez) [inline .github/workflows/dependencies-review.yml:20]:
```suggestion
          comment-summary-in-pr: true
```

File: .github/workflows/dependencies-review.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  dependency-review:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - name: 'Checkout Repository'
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: 'Dependency Review'
+        uses: actions/dependency-review-action@3b139cfc5fae8b618d3eae3675e383bb1769c019 # v4.5.0
```
Comment (ezequiel-rodriguez) [inline .github/workflows/dependencies-review.yml:17]:
```suggestion
        actions/dependency-review-action@72eb03d02c7872a771aacd928f3123ac62ad6d3a # v4.3.3
```

**Review by ezequiel-rodriguez — DISMISSED** [review]
LGTM

File: .github/workflows/dependencies-review.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  dependency-review:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - name: 'Checkout Repository'
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: 'Dependency Review'
+        actions/dependency-review-action@72eb03d02c7872a771aacd928f3123ac62ad6d3a # v4.3.3
```
Comment (ezequiel-rodriguez) [inline .github/workflows/dependencies-review.yml:17]:
```suggestion
        uses: actions/dependency-review-action@72eb03d02c7872a771aacd928f3123ac62ad6d3a # v4.3.3
```

File: .github/workflows/dependencies-review.yml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    permissions:
+      pull-requests: write
+    steps:
+      - name: 'Checkout Repository'
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: 'Dependency Review'
+        uses: actions/dependency-review-action@72eb03d02c7872a771aacd928f3123ac62ad6d3a # v4.3.3
+        with:
+          fail-on-severity: high
```
Comment (diego-jeronymo) [inline .github/workflows/dependencies-review.yml:19]: Please try it. `fail-on-severity: critical`

Comment (ezequiel-rodriguez) [inline .github/workflows/dependencies-review.yml:19]:
```suggestion
          fail-on-severity: critical
```

**Pull request merged**